lara-typescript/src/interactive-api-client/README.md

```diff
@@ -14,127 +13,55 @@ First install the package:
 
-### Hook
+### Hooks
 
-If you are using React you will instead want to the use the hook which wraps the client in a more easily used form.  To use the hook first import it:
+If you are using React you will instead want to the use the hooks which wrap the API functions in a more easily 
+used form. All the hooks start with `use` prefix. To use the hook first import it:
 
-`import { useLaraInteractiveApi } from "@concord-consortium/lara-interactive-api"`
+`import { useInteractiveState } from "@concord-consortium/lara-interactive-api"`
 
 and then call the hook using your type variables in your React component.
 
-Here is a minimal example of a hook that supports no features and has no authored or interactive state that to gets the user info:
-
-```
-import { IAuthInfo } from "@concord-consortium/lara-interactive-api";
-
-....
-
-const { getAuthInfo } = useLaraInteractiveApi({ supportedFeatures: {} });
-const [authInfo, setAuthInfo] = useState<IAuthInfo|undefined>();
-const [error, setError] = useState<any>();
+Here is a minimal example of a `useInteractiveState` hook:
 
-useEffect(() => {
-  getAuthInfo()
-    .then(setAuthInfo)
-    .catch(setError);
-}, []);
 ```
-
-and there is maximal example of a hook that supports all types of state (using the same interfaces as the client example above):
-
-```
-const { interactiveState, setInteractiveState } = useLaraInteractiveApi<IInteractiveState, IAuthoredState, IGlobalInteractiveState>({
-  supportedFeatures: {
-    authoredState: true,
-    interactiveState: true,
-    aspectRatio: 1
-  }
-});
-
+const { interactiveState, setInteractiveState } = useInteractiveState<IInteractiveState>();
 ...
-
-const handleButtonClick = () => setInteractiveState({ numButtonPresses: interactiveState.numButtonPresses + 1 })
-
+const handleButtonClick = () => {
+  // Functional update, usually safer option:
+  setInteractiveState((prevState: InteractiveState) => ({ numButtonPresses: prevState.numButtonPresses + 1 }));
+  // Or regular update that can cause troubles when new state is based on the previous one:
+  // setInteractiveState({ numButtonPresses: interactiveState.numButtonPresses + 1 });
+}
 ...
-
 <button onClick={handleButtonClick}>{interactiveState.numButtonPresses}</button>
 ```
 
-Calling the `setAuthoredState`, `setInteractiveState` and `setGlobalInteractiveState` methods returned by the hook will both update the state held inside the hook and also send the message to Lara.  This allows you to use the update the interactive state and use it in your render result.  You also do not have to supply an `onGetInteractiveState` callback as the hook handles that for you using the current local hook state.
\ No newline at end of file
+Calling the `setInteractiveState` method returned by the hook will both update the state held inside the hook and 
+also send the message to Lara. This allows you to use the update the interactive state and use it in your render result. 
```

Comment:
This doesn't read right. Perhaps:
"This means that calling setInteractiveState will trigger a new React render of your components that use the `useInteractiveState` hook"

Reply:
Thanks, fixed in the last commit.